erpnext/accounts/doctype/payment_reconciliation/test_payment_reconciliation.py

```diff
@@ -1845,6 +1846,107 @@ def test_cr_note_payment_limit_filter(self):
 		self.assertEqual(len(pr.invoices), 1)
 		self.assertEqual(len(pr.payments), 1)
 
+	def test_reconciliation_on_closed_period_payment(self):
+		from erpnext.accounts.doctype.account.test_account import create_account
+
+		self.create_company()
+		self.create_cost_center()
+
+		# create bank account
+		parent_account = frappe.db.get_value(
+			"Account", {"company": self.company, "account_name": "Bank Accounts", "is_group": 1}, "name"
+		)
+		bank_account = create_account(
+			account_name="Bank Account",
+			account_type="Bank",
+			is_group=0,
+			company=self.company,
+			root_type="Asset",
+			report_type="Balance Sheet",
+			account_currency="INR",
+			parent_account=parent_account,
+			doctype="Account",
+		)
+
+		# create backdated fiscal year
+		create_fiscal_year(company=self.company, year_start_date="1990-04-01", year_end_date="1991-03-31")
```

Comment:
Rather than a hardcoded Fiscal year, a dynamic one would be better.

Reply:
@ruthra-kumar Thanks for your feedback, I'll update the testcase.